test/local-fs.spec.ts

```diff
@@ -106,5 +110,67 @@ describe(`the local filesystem implementation`, () => {
                 })
                 .then(() => expect(fs.loadTextFile(fileName)).to.eventually.equals(newContent));
         });
+
+        it(`ignores events from ignored dir`, () => {
+            mkdirSync(join(testPath, ignoredDir))
+            return matcher.expect([])
+        });
+
+        it(`ignores events from ignored file`, () => {
+            const [dirName] = ignoredFile.split(pathSeparator);
+            mkdirSync(join(testPath, dirName))
+            return matcher.expect([{type: 'directoryCreated', fullPath: dirName}])
+                .then(() => writeFileSync(join(testPath, ignoredFile), content))
+                .then(() => matcher.expect([]))
+        });
+
+        it(`loadDirectoryTree() ignores ignored folder and file`, () => {
+            const [dirName] = ignoredFile.split(pathSeparator);
+            const expectedStructure = {
+                name: '',
+                type: 'dir',
+                fullPath: '',
+                children: [{ name: dirName, type: 'dir', fullPath: dirName, children: []}]
+            };
+
+            mkdirSync(join(testPath, ignoredDir))
+            mkdirSync(join(testPath, dirName))
+            writeFileSync(join(testPath, ignoredFile), content)
+            return expect(fs.loadDirectoryTree()).to.eventually.deep.equal(expectedStructure)
+        });
+
+        it(`deleting ignored file - fails`, function() {
+            const [dirName] = ignoredFile.split(pathSeparator);
+            mkdirSync(join(testPath, dirName))
+            writeFileSync(join(testPath, ignoredFile), content)
+
+            return expect(fs.deleteFile(ignoredFile)).to.be.rejectedWith(Error)
+        });
+
+        it(`deleting ignored directory - fails`, function() {
+            mkdirSync(join(testPath, ignoredDir))
+
+            return expect(fs.deleteDirectory(ignoredDir)).to.be.rejectedWith(Error)
+        });
+
+        it(`loading ignored file - fails`, function() {
+            const [dirName] = ignoredFile.split(pathSeparator);
+            mkdirSync(join(testPath, dirName))
+            writeFileSync(join(testPath, ignoredFile), content)
+
+            return expect(fs.loadTextFile(ignoredFile)).to.be.rejectedWith(Error)
+        });
+
+        it(`saving ignored file - fails`, function() {
+            return expect(fs.saveFile(ignoredFile, 'foo')).to.be.rejectedWith(Error)
+        });
+
+        it(`deleting ignored file - fails`, function() {
```

Comment:
extract "deleting ignored file - fails" and "saving ignored dir - fails" tests to contract test